testutil/promrated/prometheus.go

```diff
@@ -41,6 +81,73 @@ func serveMonitoring(addr string) error {
 	return errors.Wrap(server.ListenAndServe(), "failed to serve prometheus metrics")
 }
 
+func getPubkeyToClusterInfo(ctx context.Context, promAuth string) (map[string]prometheus.Labels, error) {
+	res, err := getPromClusters(ctx, promAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode == 200 {
+		result, err := readPromJSON(ctx, res)
+		if err != nil {
+			return nil, err
+		}
+
+		keyToClusterLabels := make(map[string]prometheus.Labels)
+
+		for _, cluster := range result.Data.Result {
+			if cluster.Metric.ClusterInfo != nil && cluster.Metric.ClusterName != "" {
+				keyToClusterLabels[cluster.Metric.PubKey] = map[string]string{
+					"cluster_name":    cluster.Metric.ClusterName,
+					"cluster_hash":    cluster.Metric.ClusterHash,
+					"cluster_network": cluster.Metric.ClusterNetwork,
+					"cluster_peer":    cluster.Metric.ClusterPeer,
+					"pubkey":          cluster.Metric.PubKey,
+				}
+			}
+		}
+
+		return keyToClusterLabels, nil
+	}
+
+	return nil, errors.New("error processing prom metrics", z.Str("url", res.Request.RequestURI))
+}
+
+func getPromClusters(ctx context.Context, promAuth string) (*http.Response, error) {
+	client := new(http.Client)
+	url := fmt.Sprintf("%s/query?query=%s", promEndpoint, promQuery)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		log.Error(ctx, "HTTP Request malformed for prom query", err)
+		return nil, errors.Wrap(err, "error creating prom metrics query")
```

Comment:
wrapped error messages should be concise, otherwise it makes the final error hard to read and full of noise, suggest: `errors.Wrap(err, "new prometheus request")`

Reply:
also no need to add "failure" or "error" when wrapping, the original error usually contains that, and it is clear from the fact that you are logging an error that it is an error.